File: ios/NativeCore/Sources/NativeCore/SplitLayout.swift
import Foundation

public enum SplitLayout {
    public static func commit(_ ratio:Double)->Double { ratio.isFinite ? min(1,max(0,ratio)) : 0.5 }
    public static func resize(startRatio:Double,delta:Double,extent:Double)->Double {
        let start=commit(startRatio)
        guard extent.isFinite,extent>0,delta.isFinite else { return start }
        return commit(start+delta/extent)
    }
    public static func restore(_ ratio:Double)->Double { ratio.isFinite && ratio>0 && ratio<1 ? ratio : 0.5 }
    public static func isHorizontal(width:Double,height:Double)->Bool { width>=840 && height>=600 && width>height }
}

File: android/nativepreview/src/main/java/com/travelplaner/nativepreview/domain/SplitLayout.kt
package com.travelplaner.nativepreview.domain

object SplitLayout {
    fun commit(ratio:Double):Double = if(ratio.isFinite()) ratio.coerceIn(0.0,1.0) else 0.5
    fun resize(startRatio:Double,delta:Double,extent:Double):Double {
        val start=commit(startRatio)
        return if(extent.isFinite() && extent>0 && delta.isFinite()) commit(start+delta/extent) else start
    }
    fun restore(ratio:Double):Double = if(ratio.isFinite() && ratio>0 && ratio<1) ratio else 0.5
    fun isHorizontal(width:Double,height:Double):Boolean = width>=840 && height>=600 && width>height
}
